gitlab/base.py

```diff
@@ -51,6 +52,11 @@ class RESTObject(object):
     manager: "RESTManager"
 
     def __init__(self, manager: "RESTManager", attrs: Dict[str, Any]) -> None:
+        if not isinstance(attrs, dict):
+            raise GitlabParsingError(
+                "Attempted to initialize RESTObject with a non-dictionary value: "
```

Comment:
I think `GitlabParsingError` is fine as we use it elsewhere but my first thought would be simply https://docs.python.org/3/library/exceptions.html#TypeError :)

Just wondering if this message helps the user - the other parsing errors all just throw "Failed to parse the server message". attrs will be `server_data` except when lazily creating objects, so would adding something like `"Unexpected type passed to object: {attrs} This may indicate incorrect/malformed server response"`, or similar, help give more of a clue?

Reply:
So compared to what is the current case I do think it helps the user more. The previous `TypeError: list indices must be integers or slices, not str` was very confusing.

Could it be better? Yes.  I'm open to a better message on what to say here.

It also helps us as we are better able to help them in troubleshooting if they ask questions.

Reply:
I could change it to TypeError, but I guess I liked it to be a Gitlab specific exception. To make it clear it was a Gitlab/python-gitlab thing.